tests/rules/no-multiple-actions-in-effects.test.ts

```diff
@@ -96,112 +80,90 @@ ruleTester().run(path.parse(__filename).name, rule, {
             this.actions$.pipe(mergeMap(_ => { return [foo(), bar()] }))
                                                       ~~~~~~~~~~~~~~  [${messageId}]
           )
         }`,
-      {
-        parserOptions: {
-          project: './tsconfig.tests.eslint.json',
-        },
-        filename: 'test.ts',
-      },
     ),
     fromFixture(
       stripIndent`
+        @Injectable()
         export class Effects {
-          three$ = createEffect(() =>
+          effectNOK2$ = createEffect(() =>
             this.actions$.pipe(exhaustMap(function() { return [foo(), bar()] }))
                                                               ~~~~~~~~~~~~~~  [${messageId}]
           )
         }`,
-      {
-        parserOptions: {
-          project: './tsconfig.tests.eslint.json',
-        },
-        filename: 'test.ts',
-      },
     ),
     fromFixture(
       stripIndent`
+        @Injectable()
         export class Effects {
-          readonly ten$ = createEffect(() =>
-            this.actions$.pipe(concatMapTo([test(), baz()]))
-                                           ~~~~~~~~~~~~~~~  [${messageId}]
+          readonly effectNOK3$ = createEffect(() =>
+            this.actions$.pipe(concatMapTo([foo(), bar()]))
+                                           ~~~~~~~~~~~~~~  [${messageId}]
           )
         }`,
-      {
-        parserOptions: {
-          project: './tsconfig.tests.eslint.json',
-        },
-        filename: 'test.ts',
-      },
     ),
     fromFixture(
       stripIndent`
+        const actions = [foo(), bar()]
+        @Injectable()
         export class Effects {
-          eight$ = createEffect(() =>
+          readonly effectNOK4$ = createEffect(() =>
+            this.actions$.pipe(concatMapTo(actions))
+                                           ~~~~~~~  [${messageId}]
+          )
+        }`,
+    ),
+    fromFixture(
+      stripIndent`
+        const actions = [foo(), bar()]
+        @Injectable()
+          export class Effects {
+          effectNOK5$ = createEffect(() =>
             this.actions$.pipe(
               exhaustMap(() => {
                 return of({}).pipe(
-                  switchMap(() => [foo(), bar()]),
-                                  ~~~~~~~~~~~~~~  [${messageId}]
+                  switchMap(() => actions),
+                                  ~~~~~~~  [${messageId}]
                   catchError(() => of(bar()))
                 );
               })
             )
           )
         }`,
-      {
-        parserOptions: {
-          project: './tsconfig.tests.eslint.json',
-        },
-        filename: 'test.ts',
-      },
     ),
     fromFixture(
       stripIndent`
+        @Injectable()
         export class Effects {
-          four$ = createEffect(() =>
+          effectNOK6$ = createEffect(() =>
             this.actions$.pipe(concatMap(() => {
               let actions: Action[] = [];
               return actions;
                      ~~~~~~~  [${messageId}]
```

Comment:
An interesting behavior I've noticed is that, this:

```
return actions.length > 0 ? actions : of(foo());
```

doesn't report as we may expect, **but** this:


```
return actions.length > 0 ? actions : null;
```

**does**.

---

That being said, to provide a full support for all possible cases like `?.`, `??`, `? :`, `||`, `&&` would increase the complexity and I'm not too sure it brings much value as no one complains until now.

Reply:
Weird, I've taken a quick look but it's a mystery to me. 
 For now, let's keep it like thi

Reply:
Yeah, I didn't dig too much into this, but maybe it's a problem with `couldBeType`.

Reply:
@rafaelss95 fyi, I updated the comment...  you're too fast for me 😂

Reply:
👀 it actually handles it correct - https://github.com/cartant/tsutils-etc/blob/master/source/could-be-type.ts#L30
The problem here, is that we should import `of` from `rxjs`.
Otherwise its type will just be `any` and thus isn't a union.

Reply:
I keep missing those imports within those strings 🤦

Reply:
You and me both...
Just a random thought, would it make sense to:

- create a seperate file for each case and divide them into a valid and invalid subfolder
- build the spec dynamically by reading those files and adding them as cases?

That way the individual file could be typechecked.

Reply:
You mean something like what [`tslint` used to do](https://github.com/palantir/tslint/blob/master/test/rules/adjacent-overload-signatures/ignore-accessors/test.ts.lint)? I think it's a fantastic idea. Just to clarify: would we use this approach only for rules with type-checking or...?

Reply:
@rafaelss95 I don't think we'll be able to pull it of straight away.
I didn't think of the messageId and the rules that have an output 😅

Reply:
Yes, that's not trivial on a second thought 😆